lib/create-contentful-api.ts

```diff
@@ -42,11 +45,13 @@ export default function createClientApi({ http }: { http: AxiosInstance }) {
     getSpaces: function getSpaces(
       query: QueryOptions = {}
     ): Promise<Collection<Space, SpaceProps>> {
-      return endpoints.space
-        .getMany(http, {
-          query: createRequestConfig({ query: query }).params,
+      return adapter
+        .makeRequest({
+          entityType: 'Space',
+          action: 'getMany',
```

Comment:
Maybe we can make this an `enum`?

Reply:
In general I always try to stay away from enums as they are not vanilla JavaScript and typescript adds a ton of extra code.

Due to the overloadings of `makeRequest`, we have type safety here that would catch typos. Also, we have very specific actions here like: `processForLocale`, `createFromFiles` or `getManyForOrganization`. If it was only create/update/save/delete, an enum would probably be alright.

Reply:
I'd also be in favor of an enum. Also: why do we want to have this specific actions? Assets are always processed for a locale, I don't know why I would add `ForOrganization` when getting many (parent entity is known, as in AppDefinition belongs to and org vs Entry belonging to an env), I don't know what `createFromFiles` refers to.

Reply:
I was looking through all existing methods of the plain client (https://github.com/contentful/contentful-management.js/blob/master/lib/plain/plain-client.ts) and assumed (without looking at the actual implementation) each method would eventually be one action.

I was referring to `asset.createFromFiles`, `asset.processForAllLocales` and `usage.getManyForOrganization`.

I would be happy to use enums if all actions are available for all entity types. But in this case, the actions are very specific and sometimes only apply to a single entity type. Also, I don't see the benefit of using an enum here as these actions are all library internal and we can achieve the same level of type safety by using union types.